### PR TITLE
fix: [DEL-3240]: secrets used in a delegateProfile can be inside sing…

### PR DESCRIPTION
**Pull request by arpit3107**
…le quotes also

You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/30514)

### CODE REVIEW

**Comment by arpit3107** [comment]
trigger messagecheck

File: 400-rest/src/main/java/software/wings/service/impl/DelegateProfileServiceImpl.java [hunk 1]
```diff
@@ -552,12 +553,32 @@ public boolean identifierExists(String accountId, DelegateEntityOwner owner, Str
...
+      if (!secret.equals("") && !secretService.getAccountScopedSecretByName(accountId, secret).isPresent()) {
         secretsName.add(secret);
       }
     }
     return secretsName;
   }
+
+  private String getSecretNameFromMatcher(Matcher matcher) {
+    String secret = "";
+    try {
```
Comment (abhinav-harness) [inline 400-rest/src/main/java/software/wings/service/impl/DelegateProfileServiceImpl.java:566]: make it null

File: 400-rest/src/main/java/software/wings/service/impl/DelegateProfileServiceImpl.java [hunk 2]
```diff
@@ -552,12 +553,32 @@ public boolean identifierExists(String accountId, DelegateEntityOwner owner, Str
...
+      log.debug("Exception occurred during parsing secret name inside double quotes", e);
+    }
+
+    try {
+      secret = matcher.group(0).substring(matcher.group(0).indexOf('\'') + 1, matcher.group(0).lastIndexOf('\''));
+    } catch (Exception e) {
+      log.debug("Exception occurred during parsing secret name inside single quotes", e);
+    }
+    if (secret.equals("")) {
+      log.error("Unable to parse some secret name used in the delegate profile");
```
Comment (abhinav-harness) [inline 400-rest/src/main/java/software/wings/service/impl/DelegateProfileServiceImpl.java:578]: do null check here instead

**Review by abhinav-harness — CHANGES_REQUESTED** [review]

File: 400-rest/src/main/java/software/wings/service/impl/DelegateProfileServiceImpl.java [hunk 3]
```diff
@@ -552,12 +553,32 @@ public boolean identifierExists(String accountId, DelegateEntityOwner owner, Str
     }
     Matcher matcher = secretNamePattern.matcher(script);
     while (matcher.find()) {
-      String secret =
-          matcher.group(0).substring(matcher.group(0).indexOf("\"") + 1, matcher.group(0).lastIndexOf("\""));
-      if (!secretService.getAccountScopedSecretByName(accountId, secret).isPresent()) {
+      String secret = getSecretNameFromMatcher(matcher);
+      if (!secret.equals("") && !secretService.getAccountScopedSecretByName(accountId, secret).isPresent()) {
```
Comment (abhinav-harness) [inline 400-rest/src/main/java/software/wings/service/impl/DelegateProfileServiceImpl.java:557]: why are we doing empty check here?

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30514)* status: :shipit: complete! ✓

**Pull request merged**